organisations/models.py

```diff
@@ -110,15 +113,69 @@ class Meta:
             ),
         ]
 
-    @transaction.atomic
-    def delete(self, *args, **kwargs):
-        self.delete_related_p_event_contact_info()
-        return super().delete(*args, **kwargs)
+    def _delete_related_objects(self, using=None, keep_parents=False):
+        """
+        Deletes all related objects of the current instance.
+
+        This method replicates the logic of Django's internal delete mechanism
+        to delete related objects without deleting the instance itself.
+
```

Comment:
Linking to the source on which this function is based would bring more clarity/transparency:
```suggestion

        Based on Django's Model.delete method:
        https://github.com/django/django/blob/4.2.20/django/db/models/base.py#L1123-L1132

```